manager/dispatcher/dispatcher.go

```diff
@@ -387,6 +393,19 @@ func (d *Dispatcher) updateNode(nodeID string, description *api.NodeDescription)
 	return nil
 }
 
+// gets the node IP from the context of a grpc call
+func nodeIPFromContext(ctx context.Context) (string, error) {
```

Comment:
Since raft does this as well, this might be a useful utility function for the `ca` package. But that could be a different PR.